cmd/nerdctl/exec.go

```diff
@@ -280,3 +277,44 @@ func execShellComplete(cmd *cobra.Command, args []string, toComplete string) ([]
 	}
 	return nil, cobra.ShellCompDirectiveNoFileComp
 }
+
+// replaceOrAppendEnvValues returns the defaults with the overrides either
+// replaced by env key or appended to the list
+// FYI: https://github.com/containerd/containerd/blob/698622b89a053294593b9b5a363efff7715e9394/oci/spec_opts.go#L186-L222
+func replaceOrAppendEnvValues(defaults, overrides []string) []string {
```

Comment:
Can be moved to a separate pkg, with unit tests

Reply:
How about create a `cmd/nerdctl/util.go & util_test.go` and put there? I think the logic is quite `cmd` specific. Otherwise maybe we can create a `pkg/flagutil` which contains all `flag` related logic.